File: FilFeed/src/main/java/com/phicdy/filfeed/alarm/BootReceiver.java
package com.phicdy.filfeed.alarm;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    public BootReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent != null && intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            AlarmManagerTaskManager.setNewAlarm(context);
        }
    }
}

File: domain/src/main/java/com/phicdy/mycuration/domain/alarm/BootReceiver.kt
package com.phicdy.mycuration.domain.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.phicdy.mycuration.data.preference.PreferenceHelper

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent?) {
        if (intent != null && intent.action == Intent.ACTION_BOOT_COMPLETED) {
            AlarmManagerTaskManager(context).run {
                setNewAlarm(PreferenceHelper.autoUpdateIntervalSecond)
                setFixUnreadCountAlarm()
            }
        }
    }
}
